esrally/track/track.py

```diff
@@ -701,44 +701,45 @@ class OperationType(Enum):
     CompositeAgg = (18, AdminStatus.No, serverless.Status.Public)
     WaitForCurrentSnapshotsCreate = (19, AdminStatus.No, serverless.Status.Internal)
     Downsample = (20, AdminStatus.No, serverless.Status.Internal)
+    Esql = (21, AdminStatus.No, serverless.Status.Internal)
```

Comment:
This should be `serverless.Status.Public` if we expect users _without_ operator privileges to be able to use it, which I think we do.

Reply:
Thanks, Brad! I chose `serverless.Status.Internal` because we haven't exposed this API in the serverless yet. However, I think it should be okay to use `serverless.Status.Public` here. I have adjusted this in https://github.com/elastic/rally/pull/1791/commits/a053ecce9e85cbbca795b5bac7e2925461840cf8.

Reply:
Oh, right. All this enum does is change the behaviour of Rally based on whether it is being executed against a serverless cluster with or without operator privileges. 

If you set 'Internal', then this operation is skipped in any track you run without operator privileges. If you set 'Public', then Rally will execute the operation regardless of the user's privileges. If the API isn't exposed yet, it means that this operation will fail.

Reply:
Ok, I've reverted this back to `Internal`. We will change this to Public when we expose ESQL in serverless.

Reply:
Sorry for the back and forth here, but changing the enum could/may require us to cut a new Rally release, which I don't think is a huge problem, but it may be less work to just merge this as `Public` and accept that the operations will fail until we expose the API.

@gbanasiak - you're probably the person with the most experience around cutting releases and serverless changes, WDYT?

Reply:
Today `_query` endpoint and thus the new `esql` operation have `serverless.Status.Blocked` because the endpoint is available neither to serverless operator nor to a normal user. I would prefer to start with `Blocked` and come back and switch it to `Public` once API becomes exposed in serverless. We should be ready to cut a new Rally release quickly _just_ for this.

Reply:
@gbanasiak That makes sense. I pushed https://github.com/elastic/rally/pull/1791/commits/fc7d959e10784dcae2c42f8a04b04ff4c9d9432b to use `Blocked`. Thank you!